lib/cartopy/mpl/gridliner.py

```diff
@@ -53,21 +53,11 @@
 )
 
 
-def _fix_lons(lons):
-    """
-    Fix the given longitudes into the range ``[-180, 180]``.
-
-    """
-    lons = np.array(lons, copy=False, ndmin=1)
-    fixed_lons = ((lons + 180) % 360) - 180
-    # Make the positive 180s positive again.
-    fixed_lons[(fixed_lons == -180) & (lons > 0)] *= -1
-    return fixed_lons
-
-
 def _lon_hemisphere(longitude):
     """Return the hemisphere (E, W or '' for 0) for the given longitude."""
-    longitude = _fix_lons(longitude)
+    # Wrap the longitude to the range -180 to 180, keeping positive 180s
+    lon_wrapped = ((longitude + 180) % 360) - 180
+    longitude = longitude if (longitude > 0 and lon_wrapped == -180) else lon_wrapped
```

Comment:
This would produce 180+n*360 for positives; do we not expect such high numbers? Otherwise:
```suggestion
    longitude = 180 if (longitude > 0 and lon_wrapped == -180) else lon_wrapped
```

Reply:
Good call. I think it just didn't matter because we are only verifying if longitude > 0 below, so the check would produce the same result. But this makes more sense, so taking it.

Reply:
Oh, I forgot the specific value isn't used for anything right now, but this is probably simpler to understand at least.